src/k8s/cmd/k8s/k8s_join_cluster.go

```diff
@@ -63,6 +83,8 @@ var (
 func init() {
 	joinNodeCmd.Flags().StringVar(&joinNodeCmdOpts.name, "name", "", "The name of the joining node. defaults to hostname")
 	joinNodeCmd.Flags().StringVar(&joinNodeCmdOpts.address, "address", "", "The address (IP:Port) on which the nodes REST API should be available")
+	joinNodeCmd.Flags().BoolVar(&joinNodeCmdOpts.waitReady, "wait-ready", false, "If set, the command blocks until the node is fully functional.")
```

Comment:
This is not good. For the user a Ready node means a k8s node that has registered with the API server as is reporting "Ready". The fact we are using k8sd and dqlite here does not concern the user. The event of the node registering with dqlite means nothing to the user.

We should indeed wait for the node to register with dqlite and if it does not we should report it as a failure.

Reply:
Ok, then let's drop this flag altogether.
The join command will wait for dqlite to come up and we have a `--timeout` flag if the user don't want to wait for too long.
If the user wants to wait for the node to be ready he needs to fallback to `k8s kubectl wait ...` (`k8s status --wait-ready` only waits for ONE node to be ready - not a particular one).

cc @neoaggelos